### PR TITLE
Support computed columns in SQLAlchemy ORM

### LINKED ISSUE #473: Support generated columns
Using the SQLAlchemy ORM mapper, in PostgreSQL, you can define a table with a generated column. Which is essential for generating partitioned time-series tables. Using `crate-python`, `Computed` isn't supported.

---

Using PostgreSQL, this script generated a table with a generated column:
```python
from sqlalchemy.orm import declarative_base
import sqlalchemy as sa

engine = sa.create_engine("postgresql+psycopg2://postgres:mysecretpassword@localhost:5432/test")

Base = declarative_base(bind=engine)

class Rawdata(Base):
  __tablename__ = 'test'
  __table_args__ = {
    'postgresql_partition_by': 'partid',
  }
  ts = sa.Column(sa.TIMESTAMP, primary_key=True)
  partid = sa.Column(sa.TIMESTAMP, sa.Computed("DATE_TRUNC('day', ts)"), primary_key=True)
  
  
Rawdata.__table__.create(engine)
```
Which generates:
```sql
CREATE TABLE test (
	ts TIMESTAMP WITHOUT TIME ZONE NOT NULL, 
	partid TIMESTAMP WITHOUT TIME ZONE GENERATED ALWAYS AS (DATE_TRUNC('day', ts)) STORED NOT NULL, 
	PRIMARY KEY (ts, partid)
)
 PARTITION BY partid
```

The `crate-python` adapted script:
```python
from sqlalchemy.orm import declarative_base
from crate.client.sqlalchemy import types
import sqlalchemy as sa

engine = sa.create_engine('crate://')
Base = declarative_base(bind=engine)

class Rawdata(Base):
  __tablename__ = 'test'
  __table_args__ = {
    'crate_partitioned_by': 'partid',
  }
  ts = sa.Column(sa.TIMESTAMP, primary_key=True)
  partid = sa.Column(sa.TIMESTAMP, sa.Computed("DATE_TRUNC('day', ts)"), primary_key=True)


Rawdata.__table__.create(engine)
```

Generates this table, which lacks the generated column:
```sql
CREATE TABLE test (
	ts TIMESTAMP, 
	partid TIMESTAMP, 
	PRIMARY KEY (ts, partid)
) PARTITIONED BY (partid)
```

### PR DESCRIPTION
**Pull request by hammerhead**
## Summary of the changes / Why this is an improvement
Fixes #473 

## Checklist

 - [X] [CLA](https://crate.io/community/contribute/cla/) is signed

### CODE REVIEW

File: src/crate/client/sqlalchemy/tests/create_table_test.py [hunk 1]
```diff
@@ -97,6 +97,28 @@
              ') CLUSTERED BY (p)\n\n'),
             ())
 
+    def test_with_computed_column(self):
+        class DummyTable(self.Base):
```
[the remark this answers is not in the document]
Comment (amotl) [reply src/crate/client/sqlalchemy/tests/create_table_test.py:101]: We already reported this to https://github.com/github/codeql/issues/11407.

File: src/crate/client/sqlalchemy/tests/create_table_test.py [hunk 2]
```diff
@@ -97,6 +97,28 @@
...
+        fake_cursor.execute.assert_called_with(
+            ('\nCREATE TABLE t (\n\t'
+             'ts LONG, \n\t'
+             'p LONG GENERATED ALWAYS AS (date_trunc(\'day\', ts)), \n\t'
+             'PRIMARY KEY (ts)\n'
+             ')\n\n'),
+            ())
+
+    def test_with_virtual_computed_column(self):
+        class DummyTable(self.Base):
```
[the remark this answers is not in the document]
Comment (amotl) [reply src/crate/client/sqlalchemy/tests/create_table_test.py:115]: I've dismissed this notice, we already reported it to https://github.com/github/codeql/issues/11407.

**Review by BaurzhanSakhariev — APPROVED** [review]
👍

**Review by amotl — CHANGES_REQUESTED** [review]
Hi Niklas,

thank you very much for this excellent contribution. Can I humbly ask you to add a short section about the usage of this feature to the documentation at [docs/sqlalchemy.rst](https://github.com/crate/crate-python/blob/master/docs/sqlalchemy.rst)?

I think another subsection within the _Table definition_ section [^1] would be the right place, just like the "\`\`_id\`\` as primary key" section. While being at it, feel free to add corresponding anchors like `.. _table-definition:` to those subsections as well.

I think it will be enough to just demonstrate the example from the test case there, showing off the entity schema together with a few words of prose.
```python
class DummyTable(self.Base):
    __tablename__ = 't'
    ts = sa.Column(sa.BigInteger, primary_key=True)
    p = sa.Column(sa.BigInteger, sa.Computed("date_trunc('day', ts)"))
```

Thank you in advance and with kind regards,
Andreas.

[^1]: https://github.com/crate/crate-python/blob/master/docs/sqlalchemy.rst#id10

**Review by amotl — APPROVED** [review]
👍

**Review by proddata — APPROVED** [review]

**Pull request merged**